src/verify.c

```diff
@@ -108,11 +117,44 @@ static void print_help(const char *name)
 	fprintf(stderr, "\n");
 }
 
+static bool compile_whitelist(const char *regex)
+{
+	int errcode;
+
+	if (picky_whitelist) {
+		regfree(picky_whitelist);
+	} else {
+		picky_whitelist = calloc(1, sizeof(*picky_whitelist));
+		if (!picky_whitelist) {
+			fprintf(stderr, "out of memory\n");
+			return false;
+		}
+	}
+
+	errcode = regcomp(picky_whitelist, regex, REG_NOSUB|REG_EXTENDED);
+	if (errcode) {
+		char *buffer;
+		size_t len;
+		len = regerror(errcode, picky_whitelist, NULL, 0);
+		buffer = malloc(len);
+		if (!buffer) {
+			fprintf(stderr, "out of memory\n");
+			return false;
+		}
+		regerror(errcode, picky_whitelist, buffer, len);
+		fprintf(stderr, "Invalid --picky-whitelist=%s: %s\n", regex, buffer);
+		return false;
+	}
+	return true;
+}
+
 static bool parse_options(int argc, char **argv)
 {
 	int opt;
-
-	while ((opt = getopt_long(argc, argv, "hxnItNbm:p:u:P:c:v:fYiF:qS:C:", prog_opts, NULL)) != -1) {
+	if (!compile_whitelist(picky_whitelist_default)) {
```

Comment:
You don't need to compile pick_whitelist_default. Change the name to say picky_whitelist_src,

in the 'w' case, assign optarg to picky_whitelist_src

move the compile of the regular expression until after the argument parsing has been completed.

Reply:
Okay.